### PR TITLE
wallet2: faster output and key image import/export

### PR DESCRIPTION
**Pull request by moneromooo-monero**

### CODE REVIEW

File: src/simplewallet/simplewallet.cpp [hunk 1]
```diff
@@ -4468,6 +4468,10 @@ bool simple_wallet::show_incoming_transfers(const std::vector<std::string>& args
       success_msg_writer() << tr("No incoming unavailable transfers");
     }
   }
+  else
+  {
+    success_msg_writer() << boost::format("Found %u/%u transfers") % transfers_found % m_wallet->get_num_transfer_details();
```
Comment (xiphon) [inline src/simplewallet/simplewallet.cpp]:
```suggestion
    success_msg_writer() << boost::format("Found %u/%u transfers") % transfers_found % transfers.size();
```

**Comment by danrmiller** [comment]
With older compilers and older boost I get this error that doesn't happen on newer systems:

https://build.getmonero.org/builders/monero-static-osx-10.11/builds/5778/steps/test/logs/stdio

```
In file included from /Users/buildbot/slave/monero-static-osx-10_11/build/tests/fuzz/cold-outputs.cpp:30:
In file included from /Users/buildbot/slave/monero-static-osx-10_11/build/contrib/epee/include/file_io_utils.h:32:
In file included from /usr/local/include/boost/filesystem/path.hpp:28:
In file included from /usr/local/include/boost/iterator/iterator_facade.hpp:28:
In file included from /usr/local/include/boost/type_traits/is_pod.hpp:15:
In file included from /usr/local/include/boost/type_traits/is_scalar.hpp:12:
/usr/local/include/boost/type_traits/is_arithmetic.hpp:18:55: fatal error: recursive template instantiation exceeded maximum depth of 256
struct is_arithmetic : public integral_constant<bool, is_integral<T>::value || is_floating_point<T>::value> {};
                                                      ^
```

**Comment by moneromooo-monero** [comment]
Does it work with -ftemplate-depth=280 ?

**Comment by stoffu** [comment]
Something seems not right, here's an example cold signing scenario:

1. A watch-only wallet receives an incoming transfer of 10.0
```
Height 190949, txid <e7fecb58eb5bf62e4da6d9d4f1e1fa94c6697d91780697db093d56580df36e91>, 10.000000000000, idx 0/0

[wallet 56sY35]: show_transfers
  190949     in       2018-10-26 13:35:30      10.000000000000 e7fecb58eb5bf62e4da6d9d4f1e1fa94c6697d91780697db093d56580df36e91 0000000000000000 0 - 

[wallet 56sY35]: incoming_transfers verbose
               amount   spent    unlocked  ringct    global index                                                               tx id      addr index                                                              pubkey                                                           key image
      10.000000000000       F      locked  RingCT          644844  <e7fecb58eb5bf62e4da6d9d4f1e1fa94c6697d91780697db093d56580df36e91>               0  <fe5f319a1b90507e12266d5c70ad9109192f4a493df90d0a4039b4cf8807c46f>    ????????????????????????????????????????????????????????????????
Found 1/1 transfers
```

2. Export outputs to the offline wallet and import key images from it
```
[wallet 56sY35]: export_outputs out
Outputs exported to out

[wallet 56sY35]: import_key_images ki
Signed key images imported to height 190949, 0.000000000000 spent, 10.000000000000 unspent

[wallet 56sY35]: incoming_transfers verbose
               amount   spent    unlocked  ringct    global index                                                               tx id      addr index                                                              pubkey                                                           key image
      10.000000000000       F      locked  RingCT          644844  <e7fecb58eb5bf62e4da6d9d4f1e1fa94c6697d91780697db093d56580df36e91>               0  <fe5f319a1b90507e12266d5c70ad9109192f4a493df90d0a4039b4cf8807c46f>    9eda4d057db68caa42b01172e80cc29bb813c2d1eab66f67a9a7d1a916409c37
Found 1/1 transfers

[wallet 56sY35]: balance
Currently selected account: [0] Primary account
Tag: (No tag assigned)
Balance: 10.000000000000, unlocked balance: 10.000000000000
```

3. Initiate an outgoing transfer of 5.0
```
[wallet 56sY35]: transfer 57hGLsqr6eLjUDoqWwP3Ko9nCJ4GFN5AyezdxNXwpa1PMt6M4AbsBgcHH21hVe2MJrLGSM9C7UTqcEmyBepdhvFE4eyW3Kd 5
No payment id is included with this transaction. Is this okay?  (Y/Yes/N/No): y

Transaction 1/1:
Spending from address index 0
Sending 5.000000000000.  The transaction fee is 0.000314030000
Is this okay?  (Y/Yes/N/No): y
Unsigned transaction(s) successfully written to file: unsigned_monero_tx
```

4. Sign the tx with the offline wallet

5. Submit the tx from the watch-only wallet
```
[wallet 56sY35]: submit_transfer
Loaded 1 transactions, for 10.000000000000, fee 0.000314030000, sending 5.000000000000 to 57hGLsqr6eLjUDoqWwP3Ko9nCJ4GFN5AyezdxNXwpa1PMt6M4AbsBgcHH21hVe2MJrLGSM9C7UTqcEmyBepdhvFE4eyW3Kd, 4.999685970000 change to 56sY358Fqy8TnUVajKvL4bGK4KitjggPmL1fy1KKqhieCoPRLPbNE9LPoDVDUx5xwkW6ijd3kBKaGbHsdnutQsPAHpn9SMw, with min ring size 11, no payment ID. 1 key images to import. Is this okay? (Y/Yes/N/No): y
Transaction successfully submitted, transaction <8699c2a7e3f79a079cd662b7f4456fa0bc99b1a868d170b328242e5077e5eac2>
You can check its status by using the `show_transfers` command.
```

6. Have the tx confirmed and receive the change
```
Height 190961, txid <8699c2a7e3f79a079cd662b7f4456fa0bc99b1a868d170b328242e5077e5eac2>, 4.999685970000, idx 0/0
Height 190961, txid <8699c2a7e3f79a079cd662b7f4456fa0bc99b1a868d170b328242e5077e5eac2>, spent 10.000000000000, idx 0/0

[wallet 56sY35]: show_transfers
  190949     in       2018-10-26 13:35:30      10.000000000000 e7fecb58eb5bf62e4da6d9d4f1e1fa94c6697d91780697db093d56580df36e91 0000000000000000 0 - 
  190961    out       2018-10-26 13:48:18       5.000000000000 8699c2a7e3f79a079cd662b7f4456fa0bc99b1a868d170b328242e5077e5eac2 0000000000000000 0.000314030000 57hGLsqr6eLjUDoqWwP3Ko9nCJ4GFN5AyezdxNXwpa1PMt6M4AbsBgcHH21hVe2MJrLGSM9C7UTqcEmyBepdhvFE4eyW3Kd: 5.000000000000 0 - 

[wallet 56sY35]: incoming_transfers verbose
               amount   spent    unlocked  ringct    global index                                                               tx id      addr index                                                              pubkey                                                           key image
      10.000000000000       T    unlocked  RingCT          644844  <e7fecb58eb5bf62e4da6d9d4f1e1fa94c6697d91780697db093d56580df36e91>               0  <fe5f319a1b90507e12266d5c70ad9109192f4a493df90d0a4039b4cf8807c46f>    9eda4d057db68caa42b01172e80cc29bb813c2d1eab66f67a9a7d1a916409c37
       4.999685970000       F      locked  RingCT          644857  <8699c2a7e3f79a079cd662b7f4456fa0bc99b1a868d170b328242e5077e5eac2>               0  <74da655bb30cb8202cb6790a044f51a602f35f58007d95ba219bbf3d6931e57e>    ????????????????????????????????????????????????????????????????
Found 2/2 transfers
```

7. Export outputs and import key images again
```
[wallet 56sY35]: export_outputs out
File out already exists. Are you sure to overwrite it? (Y/Yes/N/No): y
Outputs exported to out

[wallet 56sY35]: import_key_images ki
Signed key images imported to height 190949, 0.000000000000 spent, 10.000000000000 unspent

[wallet 56sY35]: incoming_transfers verbose
               amount   spent    unlocked  ringct    global index                                                               tx id      addr index                                                              pubkey                                                           key image
      10.000000000000       F    unlocked  RingCT          644844  <e7fecb58eb5bf62e4da6d9d4f1e1fa94c6697d91780697db093d56580df36e91>               0  <fe5f319a1b90507e12266d5c70ad9109192f4a493df90d0a4039b4cf8807c46f>    9eda4d057db68caa42b01172e80cc29bb813c2d1eab66f67a9a7d1a916409c37
       4.999685970000       F      locked  RingCT          644857  <8699c2a7e3f79a079cd662b7f4456fa0bc99b1a868d170b328242e5077e5eac2>               0  <74da655bb30cb8202cb6790a044f51a602f35f58007d95ba219bbf3d6931e57e>    7d01f1a56bae98b6af3db0edbb5672f652635f12e97400a3baa940ce37dfdda4
Found 2/2 transfers

[wallet 56sY35]: show_transfers
  190949     in       2018-10-26 13:35:30      10.000000000000 e7fecb58eb5bf62e4da6d9d4f1e1fa94c6697d91780697db093d56580df36e91 0000000000000000 0 - 
  190961    out       2018-10-26 13:48:18       5.000000000000 8699c2a7e3f79a079cd662b7f4456fa0bc99b1a868d170b328242e5077e5eac2 0000000000000000 0.000314030000 57hGLsqr6eLjUDoqWwP3Ko9nCJ4GFN5AyezdxNXwpa1PMt6M4AbsBgcHH21hVe2MJrLGSM9C7UTqcEmyBepdhvFE4eyW3Kd: 5.000000000000 0 - 

[wallet 56sY35]: balance
Currently selected account: [0] Primary account
Tag: (No tag assigned)
Balance: 14.999685970000, unlocked balance: 10.000000000000
```

Now, the balance is wrong since the first incoming transfer is incorrectly flagged as unspent.

**Comment by xiphon** [comment]
Regarding the PR

This PR breaks some unit_tests, and I don't like the idea of disabling the tests instead of fixing them.

**Comment by moneromooo-monero** [comment]
Do you mean the one I took out, or other ones ?

**Comment by xiphon** [comment]
The ones that the commit disables.

**Comment by moneromooo-monero** [comment]
I think I found the problem, will fix.

**Comment by moneromooo-monero** [comment]
These tests don't actually test the code works, they test whether it does the same thing as what it previously did, even when the change is intended. Since there is no tool to update them, it's a matter of going through a binary file manually to fish the expected data and write them to the source, which is a waste of time.

**Comment by moneromooo-monero** [comment]
Actually, I suppose I could copy the old structure and load that as before.

**Pull request merged**